python/tank/descriptor/io_descriptor/appstore.py

```diff
@@ -753,12 +753,14 @@ def __create_sg_app_store_connection(self):
                 else:
                     raise
 
-            log.debug("Connecting to %s..." % constants.SGTK_APP_STORE)
+            app_store = os.environ.get("SGTK_APP_STORE", constants.SGTK_APP_STORE)
+
+            log.debug("Connecting to %s..." % app_store)
             # Connect to the app store and resolve the script user id we are connecting with.
             # Set the timeout explicitly so we ensure the connection won't hang in cases where
             # a response is not returned in a reasonable amount of time.
             app_store_sg = shotgun_api3.Shotgun(
-                constants.SGTK_APP_STORE,
+                app_store,
```

Comment:
Could we modify constants.py instead so this change is picked up by anyone who wants to use the appstore URL. I know that in practice this is the only place we use it, but this will prevent us from being bitten by this change in the future.

Reply:
I can do, but is that not a violation of constants? 
The value would no longer be constant, and all other constants are constant, which is why I originally opted not to put it in there.

Reply:
Well, if we don't centralize the logic to get the value, then some code might use the constant without looking for the env var

Reply:
The value is constant once the library is loaded.

Reply:
Honestly, I would be tempted to move stuff OUT of constants. if something is used in one place, it shouldn't be centralized

And even if it's used in multiple places, I don't like the pattern of having a place where you define multiple contants.

Reply:
OK I'm hoping to look at the switching from attachment id to url stuff tomorrow. Since this has been through QA I'll leave it as is and look to implement that change as part of the other change I'm looking to make.